components/Shop/ShopTable.js

```diff
@@ -18,29 +19,73 @@ const StyledImageContainer = styled.div`
   gap: 5px;
 `;
 
-export default function ShopTable({ data, onItemClick }) {
+const StyledShopTableWrapper = styled.section`
+  height: calc(100vh - 120px);
+  overflow-y: auto;
+  padding-bottom: 60px;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`;
+
+const StyledTableRow = styled.tr`
+  &:nth-child(even) {
+    background-color: #f2e8da;
+  }
+
+  &:hover {
+    background-color: #e0d5c4;
+  }
+
+  td {
+    border: 1px solid #d3c0a3;
+  }
+`;
+
+export default function ShopTable({ data, onItemClick, category }) {
+  let typeIcon;
+
+  switch (category) {
+    case "food":
+      typeIcon = <HungerImage />;
+      break;
+    case "toy":
+      typeIcon = <ToyImage />;
+      break;
+  }
```

Comment:
Consider adding a default case to handle unexpected categories.

```suggestion
  switch (category) {
    case "food":
      typeIcon = <HungerImage />;
      break;
    case "toy":
      typeIcon = <ToyImage />;
      break;
    default:
      typeIcon = null; 
      break;
  }
```